lib/validation-strategies/openapi-v2.js

```diff
@@ -0,0 +1,741 @@
+'use strict';
+
+const url = require('url');
+const URL = url.URL;
+const should = require('should');
+const co = require('co');
+const ajv = require('ajv')({
+    allErrors: true,
+    verbose: true,
+    jsonPointers: true,
+    patternGroups: true,
+    extendRefs: true // optional, current default is to 'fail', spec behaviour is to 'ignore'
+});
+//meta: false, // optional, to prevent adding draft-06 meta-schema
+
+const common = require('./../common.js');
+const jptr = require('reftools/lib/jptr.js');
+const walkSchema = require('./../walkSchema.js').walkSchema;
+const wsGetDefaultState = require('./../walkSchema.js').getDefaultState;
+const linter = require('./../linter.js');
+
+setupAjv(ajv);
+
+const jsonSchema = require('../../schemas/json_v5.json');
+const validateMetaSchema = ajv.compile(jsonSchema);
+const openapi2Schema = require('../../schemas/openapi-2.0.json');
+const validateOpenAPI2 = ajv.compile(openapi2Schema);
+
+const dummySchema = { anyOf: {} };
+const emptySchema = {};
+
+class JSONSchemaError extends Error {
+    constructor(message, params) {
+        super(message);
+        this.errors = params.errors;
+    }
+}
+
+function setupAjv(ajv) {
+    const ajvFormats = require('ajv/lib/compile/formats.js');
+    ajv.addFormat('uriref', ajvFormats.full['uri-reference']);
+    ajv.addMetaSchema(require('ajv/lib/refs/json-schema-draft-04.json'));
+    ajv._refs['http://json-schema.org/schema'] = 'http://json-schema.org/draft-04/schema'; // optional, using unversioned URI is out of spec
+    const metaSchema = require('ajv/lib/refs/json-schema-v5.json');
```

Comment:
OpenAPI 2.0 used a version of JSON Schema based on Draft 4, so some of this Draft 5 setup is not required.